libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -879,25 +883,31 @@ class MapboxNavigation(
      * Navigation might switch to a fallback tiles version when target tiles are not available
      * and return back to the target version when tiles are loaded.
      *
+     * @param observer NavigationVersionSwitchObserver
+     *
      * @see [NavigationVersionSwitchObserver]
      */
     fun unregisterNavigationVersionSwitchObserver(observer: NavigationVersionSwitchObserver) {
         navigationVersionSwitchObservers.remove(observer)
     }
 
     /**
-     * Register a [NavigationSessionStateObserver] to be notified of the various Session states. Not publicly available
+     * Register a [NavigationSessionStateObserver] to be notified of the various Session states.
+     *
+     * @param navigationSessionStateObserver NavigationSessionStateObserver
      */
-    internal fun registerNavigationSessionObserver(
+    fun registerNavigationSessionObserver(
```

Comment:
```suggestion
    fun registerNavigationSessionStateObserver(
```

Reply:
Since the session state now also carries dynamic session ID information, we should expose a `NavigationSessionState` getter as well.